jobmanager/WorkerThread.cpp

```diff
@@ -45,6 +95,68 @@ void WorkerThread::run()
     {
         logger::error("unknown exception", "WorkerThread");
     }
+    workerThreadHandler = nullptr;
+}
+
+// return true if any new jobs were processed
+bool WorkerThread::processJobs()
+{
+    if (_backgroundJobCount == _backgroundJobs.size())
+    {
+        _backgroundJobCount = processBackgroundJobs();
+        if (_backgroundJobCount == _backgroundJobs.size())
+        {
+            // do not take on more jobs as we cannot store them if in case they are be multi-step job
+            return false;
```

Comment:
I would prefer having a vector with 512 preallocated than this fix value. WT already allocates memory in other places. To me it should preallocate a value that will work in most of times in a healthy system but allow it to grow until a bigger limit in case if we experience a spike.

Also it does not make sense for me avoid allocate memory to avoid the time of a syscall can take in exchange of delaying the job processing (which probably will cause a much bigger delay than the allocation itself)

Reply:
Suspending the job by putting it in the _backgroundJobs will actually not cost anything for the WorkerThread. Unless, it has to allocate memory.
The memory allocations we do in WT due to jobs doing that is a bit unfortunate and we try to avoid it.
However, I will make it a vector with 512 reserved slots. It will cost nothing up to that point and then a chunk will be allocated by stl.